app/packages/core/src/components/ColorModal/colorPalette/AttributeColorSetting.tsx

```diff
@@ -174,18 +218,33 @@ const AttributeColorSetting: React.FC<ColorPickerRowProps> = ({
   useEffect(() => {
     if (!values) {
       const copy = cloneDeep(fields);
+
       const idx = fields.findIndex((s) => s.path == activePath);
       if (idx > -1) {
-        copy[idx].valueColors = [defaultValue];
-        setColorScheme(false, { colorPool, fields: copy });
+        copy[idx]["valueColors"] = [defaultValue];
+        setColorScheme(false, { colorPool, fields: copy, labelTags });
       }
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [values]);
+  }, [values, activePath]);
+
+  useEffect(() => {
+    if (!labelTags?.valueColors) {
+      const copy = cloneDeep(labelTags);
+
+      copy["valueColors"] = [defaultValue];
+      setColorScheme(false, { colorPool, fields, labelTags: copy });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [labelTags?.valueColors, activePath]);
 
   // on reset, sync local state with new session values
   useEffect(() => {
-    setInput(values ?? []);
+    if (activePath === "_label_tags") {
+      setInput(labelTags?.valueColors ?? []);
+    } else {
+      setInput(values ?? []);
+    }
     // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I don't think `react-hooks/exhaustive-deps` should be disabled here. The missing dependencies seem required